packages/jobs/lib/runner/lambda.ts

```diff
@@ -0,0 +1,195 @@
+                });
+                const pvResult = await lambdaClient.send(publishVersionCommand);
+                await waitUntilPublishedVersionActive(
+                    { client: lambdaClient, maxWaitTime: envs.LAMBDA_CREATE_TIMEOUT_SECS },
+                    {
+                        FunctionName: pvResult.FunctionName,
+                        Qualifier: pvResult.Version
+                    }
+                );
+                const aResult = await lambdaClient.send(
+                    new CreateAliasCommand({
+                        FunctionName: pvResult.FunctionName,
+                        Name: envs.LAMBDA_FUNCTION_ALIAS,
+                        FunctionVersion: pvResult.Version
+                    })
+                );
+                await lambdaClient.send(
+                    new PutProvisionedConcurrencyConfigCommand({
+                        FunctionName: pvResult.FunctionName,
+                        Qualifier: envs.LAMBDA_FUNCTION_ALIAS,
+                        ProvisionedConcurrentExecutions: node.provisionedConcurrency || envs.LAMBDA_PROVISIONED_CONCURRENCY
+                    })
+                );
+                const fleetId = node.fleetId || envs.RUNNER_LAMBDA_FLEET_ID;
+                const result = await registerWithFleet(fleetId, {
+                    nodeId: node.id,
+                    url: `${aResult.AliasArn}`
+                });
+                if (result.isErr()) {
+                    logger.error(`Error registering node ${node.id} to fleet ${fleetId}`, result.error);
+                }
+            } catch (err: any) {
+                logger.error(`Error creating function ${name}`, err);
+            }
+        });
+        return Promise.resolve(Ok(undefined));
+    }
+
+    protected getEnvironmentVariables(node: Node): Environment {
+        return {
+            Variables: {
+                NODE_ENV: envs.NODE_ENV,
+                NANGO_CLOUD: String(envs.NANGO_CLOUD),
+                RUNNER_NODE_ID: `${node.id}`,
+                PERSIST_SERVICE_URL: envs.LAMBDA_PERSIST_SERVICE_URL || '',
+                JOBS_SERVICE_URL: envs.LAMBDA_JOBS_SERVICE_URL || '',
+                PROVIDERS_URL: envs.LAMBDA_PROVIDERS_URL || '',
+                NANGO_TELEMETRY_SDK: String(envs.NANGO_TELEMETRY_SDK),
+                DD_ENV: envs.DD_ENV || '',
+                DD_SITE: envs.DD_SITE || '',
+                DD_PROFILING_ENABLED: String(node.isProfilingEnabled),
+                DD_APM_TRACING_ENABLED: String(node.isTracingEnabled),
+                DD_TRACE_ENABLED: String(node.isTracingEnabled || node.isProfilingEnabled),
+                DD_API_KEY_SECRET_ARN: envs.DD_API_KEY_SECRET_ARN || ''
+            }
+        };
+    }
+
+    async terminateFunction(node: Node): Promise<Result<void>> {
+        const name = getFunctionName(node);
+        await lambdaClient.send(
+            new DeleteFunctionCommand({
+                FunctionName: name
+            })
+        );
```

Comment:
This has a cascading effect and removes all associated resources like aliases and provisioned concurrency settings. It has no effect on running functions so is safe to perform while invocations are in flight.